### ISSUE TITLE
Feature Suggestion: Obsidian Factory Output Selection

### ISSUE THREAD

**Issue by ghost** [ReikaKalseki/Reika_Mods_Issues#1748]
The obsidian factory takes in water and lava, and outputs obsidian.  However, water and lava can produce three different materials, depending on the mixture of source blocks and flow blocks.  Is it viable to add a switch (using a screwdriver, like the gearbox) that can switch the obsidian factory between obsidian, smooth stone, and cobblestone?

**Comment by ReikaKalseki** [comment]
The CC coalescence orchid is the intended means of making the stones, and I take great care to avoid adding functions in one mod that duplicate the otherwise.

**Comment by ghost** [comment]
Despite their extremely divergent contexts (engineering vs. magic)?

The key reason I'm asking about this is that it would massively increase the OF's utility in the new skyblock component of an upcoming version of Survival Industry, which obviously makes CC impossible to use (since no world generation means no pylons).  And also because it really is intuitive that anything that mixes lava and water should be able to make all three materials.

**Comment by ReikaKalseki** [comment]
> Despite their extremely divergent contexts (engineering vs. magic)?

> The key reason I'm asking about this is that it would massively increase the OF's utility in the new skyblock component of an upcoming version of Survival Industry, which obviously makes CC impossible to use (since no world generation means no pylons)

This is a valid use case...perhaps something *only* available if CC is not installed?

**Comment by redelman** [comment]
> This is a valid use case...perhaps something only available if CC is not installed?

Personally I'm not super fond of magic mods, but I play MC with plenty of people who are, so I'd rather have a config option to disable the extra outputs on the obsidian factory than have it automatically disabled if CC is present. Forcing someone into another mod they have no interest in isn't a valid option IMO. (I'm not hating on CC, I actually quite enjoy it -- it's probably the only magic mod that I like. But I don't always use it, even if other people I play with do.)

**Comment by ReikaKalseki** [comment]
I **really** do not want to introduce any competition between the two mods.

**Comment by redelman** [comment]
I don't see how it is competition. Those who want to use the obsidian factory will go that route, those who want to use the coalesence orchid will do that. It's really no different from someone using Thermal Expansion's igneous extruder instead of the orchid. It's simply a choice. Personally I will always choose the orchid over the igneous extruder, but between the obsidian factory and the orchid would depend entirely on my goals for that world or what I am doing at the time.

Looking at it another way, anything that duplicates functionality between RoC and CC could be viewed as competition, from farming to mining, to whatever.

**Comment by ghost** [comment]
First of all, I can live with the "only if CC isn't installed" thing.  It might be a problem where bug reports are concerned though ("I can't make cobblestone anymore!  Well... yes, I did just install ChromatiCraft, why?")

However, I agree with redelman; the competition is already there, inherent in the mods' contexts.  Engineers and technology buffs will want to use RotaryCraft and its siblings for all the work.  Those who want magic will find ChromatiCraft to be the more rewarding experience.  If they work together, that's just a big bonus for those who really don't care about the comparison.  If some trivial competition exists (I mean, smoothstone and cobblestone generation is hardly going to be a sticking point with anyone), is it really going to detract from the larger contexts of the respective mods?

**Comment by ReikaKalseki** [comment]
Most magic mods, if put in a large modpack, are completely ignored in favor of the tech mods that tend to be more powerful and/or less effort, unless the pack is so magic-focused that there are no alternatives. I am determined to avert that fate with ChromatiCraft, because the majority of packs with one of my "mainline" mods have all of them.

**Comment by ghost** [comment]
Let me see if I understand this correctly.

You want ChromatiCraft to be competitive enough with RotaryCraft that people will prefer it.  To this end, you want to completely eliminate duplicate functionality so that there is no overlap.  This I can understand.

However, you believe that RotaryCraft could potentially overshadow a mod that can provide personal magical abilities, time manipulation, and teleportation because of the ability to generate stone and cobblestone from the Obsidian factory?  This I can't understand.

Some perspective might be useful here.  In SI, I am quite careful to avoid duplication of functionality.  However, the functionality I try to avoid duplicating is *major* functionality.  Farming.  Mining.  >2x Ore processing.  For things like pipes and storage, however, I generally give a pass, because they are more or less trivial things that don't affect the progression of the major mods.

**Comment by ReikaKalseki** [comment]
Cobblestone generation, given that basically everyone makes at least one, is not very "trivial".

**Comment by ghost** [comment]
As a tool for play, I agree, it's not trivial.  Like you said, everyone will make at least one, whether it's using the Igneous Extruder, Coalescence Orchid, or even a redstone clock/block breaker combo.

As a feature for a mod, however, it is *absolutely* trivial; CC has *far* more useful features that cannot be duplicated in mods like RotaryCraft, such as the transmutation flame, personal abilities, and tile accelerators.

Besides, if the video I just finished watching regarding the orchid is correct, the orchid is actually a pretty poor method of cobblestone generation; this is probably the only mod in existence (and certainly the only one I know of) that will consume a water source block for *cobble*stone generation.  Even vanilla cobblestone generators will mix water and lava flow blocks (not source blocks) to generate cobblestone, meaning that the source blocks are not consumed.

**Comment by ReikaKalseki** [comment]
The consumption of water is included as a balancer for speed - the orchid can produce more than a stack of cobblestone per half-second.

**Comment by ghost** [comment]
Once again, why is this important as a *competitive* feature of ChromatiCraft?  Anyone can make cobblestone generators using any number of methods, including vanilla ones, and it won't affect their choice (or non-choice) of ChromatiCraft (or any other mod) one single bit.  That's what I mean by trivial; not that the tool is unimportant to one's play (especially during skyblock games), but that it is completely unimportant to the choice of mods one will use.

Simply put, ChromatiCraft may be passed over because of its difficulty curve, exploration grind, or because it's a magic mod, and it may be chosen because of its advanced time manipulation, transportation, and personal abilities.  But the odds of someone choosing (or rejecting) ChromatiCraft because of the cobblestone generating ability of the coalescence orchid (or the obsidian factory) is virtually nil.

**Comment by ReikaKalseki** [comment]
> But the odds of someone choosing (or rejecting) ChromatiCraft because of the cobblestone generating ability of the coalescence orchid (or the obsidian factory) is virtually nil.

That was what the mention of the ubiquity of cobble generators was for - an indication that, given a lack of other options, it would send basically everyone into CC.

However, that caveat - a lack of other options - is important. You have a point in that basically every pack has at least one other - easier - method, and vanilla has a (non-automatable) one as well.

**Comment by ghost** [comment]
Indeed... and let's not forget that vanilla cobblegens can be automated using redstone block breakers, including the ones you have in Expanded Redstone.

As such, there's really no harm to CC in adding cobble and smooth stone support to the obsidian factory.

**Comment by Rewyn** [comment]
From what I've seen, the obsidian factory is usually used rather scarcely in the first place. Adding the ability to make cobble/smoothstone would make it somewhat more attractive to use.
It's also not quite in the entry tier for RoC, so I highly doubt anyone would choose RoC **just** for that. Chances are, most won't bother in the first place.
This whole thing with not wanting the 2 mods have competition between the 2 makes sense. Until you get to the point where someone might only want to use one of them, for either thematic reasons, or simply technical ones (not being able to run CrC, for whatever reasons as an example).
Also, being *forced* to use another mod for something quite simple might turn away people away from it in the first place.
There are also other places, where you've already duplicated functions from RoC in CrC, sometimes in a far superior way. Fluid Link Anchors for example. Or even within RoC itself, with Spillways being almost always superior to pumps, if just for ease of use. (Yes, I really don't like the way spillways work :p)

**Comment by ReikaKalseki** [comment]
I have looked into this again, and I still cannot bring myself to just make the obsidian factory a more expensive and involved copy of the TE igneous extruder.

I did have one idea, though - what if I made the factory behave differently based on ratios of water to lava, with a 1:1 ratio yielding obsidian, and something heavily biased in favor of water yielding cobblestone (possibly with interpolated probabilities in between)? That way, the "intended" usage is a bit more involved, which I like, and the option remains to use it for cobblestone by "mis-supplying" it on purpose.

**Comment by ghost** [comment]
I'm not sure I understand what you're going for.  Do you mean differences in pressure between the lava and water pipes feeding into the factory?  Or do you mean that whether it creates obsidian or cobblestone depends on which tank is full first?  Or do you mean that it will only make obsidian if the lava tank isn't receiving any more lava?  If it makes cobblestone, do you intend the process to be destructive to the lava piped in?  Or do you mean that it will generate cobblestone without affecting the lava until the water level is full, at which point an obsidian block is formed?

**Comment by ReikaKalseki** [comment]
As in the relative amounts of fluid in the two tanks.

**Comment by ReikaKalseki** [comment]
This was implemented in v20.
